src/prefect/futures.py

```diff
@@ -336,11 +336,17 @@ async def resolve_futures_to_data(
         *[
             # We must wait for the future in the thread it was created in
             from_async.call_soon_in_loop_thread(
-                create_call(future._result, raise_on_failure=False)
+                create_call(future._result, raise_on_failure=raise_on_failure)
```

Comment:
👍 but you'll need to add `raise_on_failure` as a parameter to the `resolve_futures_to_data` as well